test/integration/test___main__.py

```diff
@@ -42,30 +41,26 @@ def test_main_bad_private_data_dir():
     cmdline('run', tmpfile, '-p', 'fake')
 
     try:
-        with raises(OSError):
+        with pytest.raises(OSError):
             main()
     finally:
         os.remove(tmpfile)
 
 
-def run_role(options, private_data_dir=None, expected_rc=0):
-    try:
-        private_data_dir = private_data_dir or tempfile.mkdtemp()
-        args = ['run', private_data_dir]
-        args.extend(options)
+def run_role(options, mocker, private_data_dir, expected_rc=0):
+    args = ['run', private_data_dir]
+    args.extend(options)
+    mock_run = mocker.patch('ansible_runner.interface.run')
```

Comment:
It still feels like this should be `with mocker.patch`, because the feature of being test-scoped is now more like a bug, where anything running after `run_role` is called will (rather unexpectedly) have that patch active.

Reply:
Good point.